File: integration-test-kotlin/src/test/java/com/andrzejressel/sjs/integrationtestkotlin/SimpleDTOSerializatorTest.java
package com.andrzejressel.sjs.integrationtestkotlin;

import net.jqwik.api.ForAll;
import net.jqwik.api.Property;
import pl.andrzejressel.sjs.serializator.Serializator;

import java.util.List;

class SimpleDTOSerializatorTest extends AbstractSerializatorTest<SimpleDTO> {

    @Override
    protected Serializator<SimpleDTO> getSerializator() {
        return SimpleDTOSerializator.INSTANCE;
    }

    @Property
    public void test(@ForAll String val1, @ForAll List<Integer> val2) {
        performTest(new SimpleDTO(val1, val2));
    }

}

File: integration-test-kotlin/src/test/kotlin/com/andrzejressel/sjs/integrationtestkotlin/SimpleDTOSerializatorTest.kt
package com.andrzejressel.sjs.integrationtestkotlin

import net.jqwik.api.ForAll
import net.jqwik.api.Property
import pl.andrzejressel.sjs.serializator.Serializator

class SimpleDTOSerializatorTest : AbstractSerializatorTest<SimpleDTO>() {
    override val serializator: Serializator<SimpleDTO>
        get() = SimpleDTOSerializator.INSTANCE

    @Property
    fun test(@ForAll val1: String, @ForAll val2: List<Int>, @ForAll val3: Map<String, List<Int>>) {
        performTest(SimpleDTO(val1, val2, val3))
    }
}